core/shared/src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -40,12 +40,22 @@ trait Queue[F[_], A] { self =>
   /** Dequeue one `A` from this queue. Completes once one is ready. */
   def dequeue1: F[A]
 
+  /** Dequeue all available `A`s from this queue. Completes once any values are ready. */
+  def dequeueAvailable1: F[Vector[A]]
+
   /** Like `dequeue1` but provides a way to cancel the dequeue. */
   def cancellableDequeue1: F[(F[A], F[Unit])]
 
+  /** Like `dequeueAvailable1` but provides a way to cancel the dequeue. */
+  def cancellableDequeueAvailable1: F[(F[Vector[A]], F[Unit])]
+
   /** Repeatedly call `dequeue1` forever. */
   def dequeue: Stream[F, A] = Stream.bracket(cancellableDequeue1)(d => Stream.eval(d._1), d => d._2).repeat
 
+  /** Repeatedly call `dequeueAvailable1` forever. */
+  def dequeueAvailable: Stream[F, A] =
```

Comment:
@pchlupacek My concern with the signature you're proposing is that there's no way to dequeue *all* of the elements, unless you feed the actual queue bound to the `Pipe`.  That seems ugly to me.  Like I said, this kind of abstraction is most useful in situations where your element algebra has "natural batching", where it doesn't matter how many elements are being collapsed and has no impact on fairness or flow control.  Or another way of looking at this would be as a form of variable-width catch-up on backpressure, since processing of algebrae with natural batching *speeds up* on a per-element basis as the number of elements increases.